packages/server/lib/cloud/artifacts/artifact.ts

```diff
@@ -0,0 +1,105 @@
+import Debug from 'debug'
+import { performance } from 'perf_hooks'
+
+const debug = Debug('cypress:server:cloud:artifact')
+
+const isAggregateError = (err: any): err is AggregateError => {
+  return !!err.errors
+}
+
+export interface IArtifact {
+  reportKey: 'video' | 'screenshots' | 'protocol'
```

Comment:
we could refactor this to an enum and reference the enum strings in `Artifact` constructor but I don't think its  necessary. For readability though I would love if the reportkey enum strings were in the same order here as in the Artifact class

Reply:
If only ts enums were actually typesafe - I may convert to a more typesafe object with keyof type

Reply:
probs the object with typesafefy vs an actual ts enum is better